api-consumption/src/main/java/org/zalando/nakadi/service/subscription/StreamingContext.java

```diff
@@ -313,11 +314,15 @@ private void rebalance() {
         if (null != sessionListSubscription) {
             // This call is needed to renew subscription for session list changes.
             sessionListSubscription.getData();
-            zkClient.updateTopology(topology ->
-                    rebalancer.apply(
-                            zkClient.listSessions(),
-                            topology.getPartitions())
-            );
+            try {
+                zkClient.updateTopology(topology ->
+                        rebalancer.apply(
+                                zkClient.listSessions(),
+                                topology.getPartitions())
+                );
+            } catch (final RebalanceConflictException e) {
+                log.warn("failed to rebalance partitions: {}", e.getMessage(), e);
+            }
```

Comment:
Would not it be better to handle exception where it occurs?  I mean:
```
            zkClient.updateTopology(topology ->
                try {
                        return rebalancer.apply(
                                zkClient.listSessions(),
                                topology.getPartitions())
                } catch (final RebalanceConflictException e) {
                    log.warn("failed to rebalance partitions: {}", e.getMessage(), e);
                   return []; // Or how it is written in java
                }
            );
```

Reply:
I think it eliminates the need to figure out if the exception is actually thrown in non-modified form through the unknown code hidden by interface.